### DISCUSSION TITLE
关于新增图片的问题

### DISCUSSION THREAD

**Discussion by lincou** [lisai9093/YYS#20] [Q&A] [answered]
感谢大佬之前的回复，陆陆续续看了几篇帖子，obd的问题已经解决了。
现在就是御魂画面多了一个不显示“点击空白界面”的界面，使用御魂打手的时候回卡在这个地方若干秒等它自动跳过。
我想添加新的“ying”图片应该怎么添加呢？
有尝试过在图片文件中直接新增ying2的图片，但是无效，后面查看了代码，新增后ying2字符串后程序报错。报错显示识别不到新增的ying2，因此想问问新增图片包括代码修改需要在什么位置呢？
期待大佬的回复！！！

**Comment by lisai9093** [comment]
> 不显示“点击空白界面”的界面

这个能截个图么，我看看能不能修复。

> 因此想问问新增图片包括代码修改需要在什么位置呢？

所有截图都放在png文件夹下就好了

**Reply by lincou** [reply]
![112](https://github.com/user-attachments/assets/09511185-f75a-415f-8d17-e7c7a0fb4ef4)
![111](https://github.com/user-attachments/assets/dce019ce-215a-4192-b198-89add405cdd7)
![ying1](https://github.com/user-attachments/assets/a3cf417a-8385-43a6-8be5-253c56e8819a)
大佬这么晚了还没睡觉！
图一就是打御魂出现的图，图二是我新增的标识，代码中我也新增加了文件的名称，但是测试了之后是无效的，要等到这个界面自动结束才行。
最后想知道怎么17debug是如何调试的，虽然没学过python，但是注释解释的很清晰，代码框架看的七七八八，但是这个地方还是不知道怎么调试
致敬作者大大

**Reply by lisai9093** [reply]
> ![112](https://private-user-images.githubusercontent.com/103207305/351436673-09511185-f75a-415f-8d17-e7c7a0fb4ef4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3NTk2NjQsIm5iZiI6MTcyMTc1OTM2NCwicGF0aCI6Ii8xMDMyMDczMDUvMzUxNDM2NjczLTA5NTExMTg1LWY3NWEtNDE1Zi04ZDE3LWU3YzdhMGZiNGVmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyM1QxODI5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMWJlMzU2ZDVmYjE5ZmUwYzYzMTE3YjY0NGU4YWFkZDA0N2FlY2ExZjMwMTQwMzdhZGMxZTNmZDg1ZjRiMzA5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.zVIhIlNvQpaM8F07Tq1PCa0FiD4D4438fiThvlIOM3U) ![111](https://private-user-images.githubusercontent.com/103207305/351436260-dce019ce-215a-4192-b198-89add405cdd7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3NTk2NjQsIm5iZiI6MTcyMTc1OTM2NCwicGF0aCI6Ii8xMDMyMDczMDUvMzUxNDM2MjYwLWRjZTAxOWNlLTIxNWEtNDE5Mi1iMTk4LTg5YWRkNDA1Y2RkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyM1QxODI5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMjQ5ZjkzNjI5MDQ0NTQzOWE5OWQ5MjljZmQ5MzVkYWM4MmUzYTU0MDI3YWNlNjY1NjgzZTNjMDA5YmI3MzI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.E71s-HJFF50vsbbTbrCsBkJDXI8gQZcarVLzn4Ay_pM) ![ying1](https://private-user-images.githubusercontent.com/103207305/351435843-a3cf417a-8385-43a6-8be5-253c56e8819a.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3NTk2NjQsIm5iZiI6MTcyMTc1OTM2NCwicGF0aCI6Ii8xMDMyMDczMDUvMzUxNDM1ODQzLWEzY2Y0MTdhLTgzODUtNDNhNi04YmU1LTI1M2M1NmU4ODE5YS5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyM1QxODI5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hM2Q1NGVlZTlmYzE2MTVkMGZlY2M1ZWY0ZGY4Y2EzMzg5NDhhN2IwZGQ5MmNmMGMwNzdkMzM2YWRlNzAzZTMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NOhLu9IK_povtryEKyz2Nl3putQJvdO6VYyJKMo8Ofc) 大佬这么晚了还没睡觉！ 图一就是打御魂出现的图，图二是我新增的标识，代码中我也新增加了文件的名称，但是测试了之后是无效的，要等到这个界面自动结束才行。 最后想知道怎么17debug是如何调试的，虽然没学过python，但是注释解释的很清晰，代码框架看的七七八八，但是这个地方还是不知道怎么调试 致敬作者大大

这个问题你直接把结算界面设置下就好了：https://github.com/lisai9093/YYS/discussions/17#discussioncomment-9888671

**Reply by lisai9093** [reply]
> 最后想知道怎么17debug是如何调试的，虽然没学过python，但是注释解释的很清晰，代码框架看的七七八八，但是这个地方还是不知道怎么调试

17（debug）是用来测试截图功能的。但这个问题和截图无关，只是截图后没有识别而已。

**Reply by lincou** [reply]
请问如果个人有需要的话，如何建立新的功能并调试呢（主要是调试）？
现有的逻辑是将截图和存放在png文件🀄️的图片进行比对并识别，但是如何知道图片识别成功呢？
![image](https://github.com/user-attachments/assets/ac4547cb-c765-4c20-80ac-d24d497cad13)
想知道这一段代码如何进行调试？
致敬作者大大！

**Reply by lisai9093** [reply]
图片没有识别的话会卡住不动。
上面的代码只用把msg变成bool(1)即可
<img width="536" alt="image" src="https://github.com/user-attachments/assets/12ce9f35-9631-41a2-b664-0329cc2f58e1">

**Comment by lincou** [comment, marked as answer]
感谢作者大大回复！
因为最新丰年活动（一只怪金币经验加成），不想开虚拟机刷狗粮，因此实现了单个系统mumu开两个软件实现狗粮自由。经过了一个多小时的测试，偶尔也会发生探索异常退出情况。
思路如下：
本来想写代码实现的，但是比较复杂，问题也比较多。因此换了一种思路，特地查了adb的原理，发现通过端口之间传输数据，因此反过来想，就可以知道改windows端的端口就能实现一个adb控制一个模拟器。adb默认的端口号是5037，因此新的端口号要和原来的不同，可以使用5046端口号（不同的电脑可能会有差异）。
将action.py中的comm=[adb_path,'connect','127.0.0.1:模拟器端口号']
修改成comm=[adb_path,'-P','5046','connect','127.0.0.1:模拟器端口号']
每一处的comm=[adb_path,......]都需要修改
示例：
![1721841133931](https://github.com/user-attachments/assets/c3304551-5020-42f6-b886-b6ddc8a218ec)
![1721841243393](https://github.com/user-attachments/assets/9884bf86-8b3a-4561-8643-6cc6973b9430)
还有几个需要大家自己找，这里传图好几遍都没上来，也不多，就几个。
特别注意的是，最后一张图这里一定需要加上端口号，不然第二个软件就会控制第一个模拟器，导致两个adb对一个模拟器进行操作，这是非常危险的。
![1721841370179](https://github.com/user-attachments/assets/64d9cc8c-64cb-4742-8f1f-e00ca7f3bd0e)
致敬作者大大！非常感谢作者大大的付出！！！
最后祝大家狗粮和金币刷的愉快

**Reply by lisai9093** [reply]
感谢支持，共同学习！

**Reply by lisai9093** [reply]
我最近加入了模拟器多开的功能你可以试一试。不过我没有Mumu模拟器所以只在雷电模拟器测试过。

**Reply by lincou** [reply]
没问题！看完作者大大的代码又学到了很多新的东西，作者将思路转换成代码的效率很高，这点我还是薄弱点。
插一句，看完python的各种库功能真好用，到时候有机会可以上手试试。
多开功能今晚就测试，致敬作者大大！

**Reply by lincou** [reply]
测试基本上完成了，目前出现的问题：
使用该功能的时候，如果只开一个端口对后续没什么问题，但是如果退出重新选择我想要的第二个端口，就会报错。
报错如图
![未命名图片](https://github.com/user-attachments/assets/9527e536-d68d-4cb5-8559-17813d1f7987)
可能是mumu模拟器自带的adb的问题，因为没有雷电，因此还未测试过雷电模拟器是否有相同的问题。
致敬作者大大！

**Reply by lisai9093** [reply]
啊这个选项输入0或者1 就行，不是端口。。。我等下会再加入自动提示功能

**Reply by lisai9093** [reply]
我已经加入更多提示信息，即使输入数字超出选择范围也不会闪退了。你可以再试试～

**Reply by lincou** [reply]
致敬作者大大~
优化意见：
1.选择设备建议加上提示词，比如输入数字几或者其他的提示。
2.显示范围（数组默认从0开始）不要从0开始，从1开始即可，对应几号设备。
最后，感谢作者大大让我们用上越来越方便的工具~

**Reply by lisai9093** [reply]
已更新，感谢反馈！

**Reply by chiello0** [reply]
> 致敬作者大大~ 优化意见： 1.选择设备建议加上提示词，比如输入数字几其他的提示。 2.或者显示范围（内存默认开始从0开始）不要从0开始，从1即可，回复几号设备。 最后，非常感谢作者让我们使用上越来越方便的工具~

**Comment by chiello0** [comment]
大佬  能不能出个详细教程  按照你的操作  我连改端口命令行都进不去额   感谢致敬 @lincou

**Reply by lincou** [reply]
目前作者最新版已经不需要修改代码了，下载之后直接修改action.py中链接adb的地址（一般都在mumu模拟器的shell文件中），然后打开软件输入对应adb的端口号就行了。

**Reply by chiello0** [reply]
你是直接用木木的双开器   还是下载了两个mumu

**Reply by lincou** [reply]
mumu12版本自带的多开器

**Reply by chiello0** [reply]
![局部截取_20240814_151845](https://github.com/user-attachments/assets/4b1732fa-cee2-4504-a4f0-996a6f5cb0f3)
大佬帮忙看下在哪查看 额  感谢   或者方便的话可以有偿帮我看下Y(>_<、)Y

**Reply by chiello0** [reply]
好像可以了   直接把选择adb端口这一步  改成第二个端口   就会让选择一号还是二号   不然不会跳选择。问个别的问题   两个账号用的话桌面摆放位置有要求吗感谢

**Reply by lincou** [reply]
有偿就不需要了，秉承作者大大开源精神 哈哈哈
位置没要求，唯一的缺点就是需要多复制几个开多个软件，不能一个软件控制多个账号。

**Reply by chiello0** [reply]
感谢 都一样的
